packages/devtools_app/lib/src/standalone_ui/api/impl/vs_code_api.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:json_rpc_2/json_rpc_2.dart' as json_rpc_2;
+import 'package:meta/meta.dart';
```

Comment:
do we need to add a dep on meta? I remember an effort to remove this dependency from devtools a while back, though I can't recall why. Probably because we were using it for `@required` and some other signatures that are now provided by flutter out of the box.

Reply:
I added it to use @protected to try and avoid some of the base methods in the API classes showing up to the widget code. Unfortunately code completion still shows them (https://github.com/dart-lang/sdk/issues/35449), though it does generate a warning if you use them.

It was mostly just to enforce everything going through methods (eg. don't call `sendRequest('someString')` directly from a widget), but it's a fairly small benefit. If there's a reason to avoid the dependency (and it wasn't just removed because it was _unnecessary_), we could remove it. I tracked the change back to https://github.com/flutter/devtools/pull/3484 but I don't understand the reason - wdyt?